### PR TITLE
Quick Order List + Volume Pricing

### PR DESCRIPTION
**Pull request by sofiamatulis**
### This PR introduces Volume Pricing and Quick Order List

- **Quick order List:** a new Bulk Buying Section available for the PDP. Designs can be found here: https://www.figma.com/file/3y5RP5ET0JvqORkRFpyWmH/Variants-List-Design-Review?type=design&node-id=0-1&t=ursrFVPa2NlGVTkC-0

- **Volume Pricing** available to buyers logged in with b2b on feat prod, product page and cart. Designs: https://www.figma.com/file/AxH2jkpnLbZXPR8NbgaoaK/Volume-pricing-in-flagship-themes?type=design&node-id=2-130&t=4TqjzhFcoEMyh2av-0

Here are the PRs that were merged here (there are also some other PRs that were merged in the below PRs as well): 
- Variant List https://github.com/Shopify/dawn/pull/2627
- Vol Pricing for PDP https://github.com/Shopify/dawn/pull/2711
- UI Polishes https://github.com/Shopify/dawn/pull/2794
- Price per item in quick order list https://github.com/Shopify/dawn/pull/2792
- Feat prod price per item https://github.com/Shopify/dawn/pull/2799
- Vol pricing to cart https://github.com/Shopify/dawn/pull/2806
- UI Polishes and error fixes (all PRs can be found below)

Editor: https://os2-demo.myshopify.com/admin/themes/140237864982/editor

### CODE REVIEW

File: assets/cart.js [hunk 1]
```diff
@@ -6,6 +6,8 @@ class CartRemoveButton extends HTMLElement {
       event.preventDefault();
       const cartItems = this.closest('cart-items') || this.closest('cart-drawer-items');
       cartItems.updateQuantity(this.dataset.index, 0);
+
+      publish(PUB_SUB_EVENTS.cartUpdateStarted, { variantId: event.target.parentNode.dataset.variantId });
```
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply assets/cart.js]: Done here https://github.com/Shopify/dawn/pull/2826

**Review by nicklepine — APPROVED** [review]
tophatted golden path via the link shared on the pr description. Looks great! Functional. Any tweaks can be made subsequently as a hotfix on dawn/main.

**Comment by sofiamatulis** [comment]
> * One small thing with `Variant` names and `Variant total` values not lining up on Mobile and Tablet:
>   https://screenshot.click/Mobile-Misalignment-w-SKUs-images.png
>   https://screenshot.click/Mobile-Misalignment.png
>   https://screenshot.click/Tablet-Misalignment.png

Fixed ^ https://screenshot.click/13-35-krmv1-84bik.png
https://screenshot.click/13-36-lauaq-0kh38.png

**Comment by scottgmeadows** [comment]
**Bug**: Variant list add to cart breaks if you click it 2 short bursts.

**Repro steps:**
1 - Click + icon on a quantity picker
2 - Pause
3 - Click it a bunch of times really fast.
4 - Result: Error and unexpected quantity.

**Suggested solution:** Block quantity changes when loading the same way we do in cart. 

https://github.com/Shopify/dawn/assets/2594304/43311cd5-a9dd-41df-bf8a-c1af00e384e3

**Review by KaichenWang — APPROVED** [review]

**Comment by PaulNewton** [comment]
Is this feature demoed anywhere on the actual dawn demo store
https://theme-dawn-demo.myshopify.com ?
Either just the variant-list or the volume-pricing, or both.

**Pull request merged**